src/burnchains/burnchain.rs

```diff
@@ -915,34 +896,55 @@ impl Burnchain {
                 }
                 
                 let insert_start = get_epoch_time_ms();
-                let res = Burnchain::process_block(&mut burndb, &burnchain_config, &burnchain_block)?;
-                last_processed = Some(res);
+                let (tip, transition) = Burnchain::process_block(&mut burndb, &burnchain_config, &burnchain_block)?;
+                last_processed = (tip, Some(transition));
                 let insert_end = get_epoch_time_ms();
 
                 debug!("Inserted block {} in {}ms", burnchain_block.block_height(), insert_end - insert_start);
             }
-            Ok(last_processed.unwrap())
+            Ok(last_processed)
         });
 
         // feed the pipeline!
+        let input_headers = indexer.read_headers(start_block + 1, end_block + 1)?;
+        let mut downloader_result : Result<(), burnchain_error> = Ok(());
         for i in 0..input_headers.len() {
-            downloader_send.send(Some(input_headers[i].clone()))
-                .map_err(|_e| burnchain_error::ThreadChannelError)?;
+            info!("Downloading burnchain block {} out of {}...", start_block + 1 + (i as u64), end_block);
```

Comment:
Should this be a debug rather than an `info!`? This makes followers very chatty.

Reply:
What I really want here is some kind of progress indication that tells the user how long it'll be until they have all the blocks and can start querying the chainstate and sending transactions.  This really should be done by some kind of RPC endpoint (maybe /v2/info) that tells you how many burnchain blocks are still downloading and how many Stacks blocks (i.e. sortitions) are still downloading.  This is just meant to be a poor man's stand-in for that.

Reply:
Happy to switch back to `debug!`

Reply:
fixed in e1583e1